presto-native-execution/presto_cpp/main/PrestoTask.cpp

```diff
@@ -117,45 +118,121 @@ static void addRuntimeMetric(
-  opOut.runtimeStats.emplace(statName, prestoMetric);
-  prestoTaskStats.runtimeStats[statName] = prestoMetric;
+  opOut.runtimeStats.emplace(
+      statName, createProtocolRuntimeMetric(statName, op.spilledRows));
 
   statName =
       fmt::format("{}.{}.spilledPartitions", op.operatorType, op.planNodeId);
-  prestoMetric = createProtocolRuntimeMetric(statName, op.spilledPartitions);
-  opOut.runtimeStats.emplace(statName, prestoMetric);
-  prestoTaskStats.runtimeStats[statName] = prestoMetric;
+  opOut.runtimeStats.emplace(
+      statName, createProtocolRuntimeMetric(statName, op.spilledPartitions));
 
   statName = fmt::format("{}.{}.spilledFiles", op.operatorType, op.planNodeId);
-  prestoMetric = createProtocolRuntimeMetric(statName, op.spilledFiles);
-  opOut.runtimeStats.emplace(statName, prestoMetric);
-  prestoTaskStats.runtimeStats[statName] = prestoMetric;
+  opOut.runtimeStats.emplace(
+      statName, createProtocolRuntimeMetric(statName, op.spilledFiles));
+}
+
+static void addTaskRuntimeMetrics(
+    std::unordered_map<std::string, RuntimeMetric>& taskRuntimeStats,
+    const exec::OperatorStats& op) {
+  const std::unordered_map<std::string, uint64_t> opMetrics{
+      {"numSplits", op.numSplits},
+      {"inputBatches", op.inputVectors},
+      {"outputBatches", op.outputVectors}};
+
+  for (const auto& [name, value] : opMetrics) {
+    const auto statName =
+        fmt::format("{}.{}.{}", op.operatorType, op.planNodeId, name);
+    taskRuntimeStats.emplace(
+        statName, createVeloxRuntimeMetric(value, RuntimeCounter::Unit::kNone));
+  }
+}
+
+static void addDriverTaskRuntimeMetrics(
+    protocol::TaskStats& prestoTaskStats,
+    const exec::TaskStats& taskStats) {
+  // Task runtime metrics for driver counters.
+  std::unordered_map<std::string, uint64_t> taskMetrics{
+      {"drivers.total", taskStats.numTotalDrivers},
+      {"drivers.running", taskStats.numRunningDrivers},
+      {"drivers.completed", taskStats.numCompletedDrivers},
+      {"drivers.terminated", taskStats.numTerminatedDrivers}};
+  for (const auto it : taskStats.numBlockedDrivers) {
+    taskMetrics.emplace(
+        fmt::format("drivers.{}", exec::blockingReasonToString(it.first)),
+        it.second);
+  }
+
+  for (const auto& [name, value] : taskMetrics) {
+    prestoTaskStats.runtimeStats.emplace(
+        name, createProtocolRuntimeMetric(name, value));
+  }
+}
+
+static void processOperatorStats(
+    protocol::TaskStats& prestoTaskStats,
+    protocol::TaskState state) {
+  bool skipRuntimeStats =
+      SystemConfig::instance()->skipRuntimeStatsInRunningTaskInfo() &&
+      !isFinalState(state);
+
+  if (skipRuntimeStats) {
+    for (auto& pipelineOut : prestoTaskStats.pipelines) {
+      for (auto& opOut : pipelineOut.operatorSummaries) {
+        opOut.runtimeStats.clear();
+      }
+    }
+  } else {
+    const std::vector<std::string> prefixToExclude{"running", "blocked"};
```

Comment:
Why do we exclude these?

Reply:
This happens to the operator tab. I'm thinking of only keeping the operator particular runtime stats in the operator tab. running* and blocked* stats are common to all operators, and we can have them in the task tab for x-operator comparison.